referenceframe/frame_json.go

```diff
@@ -48,21 +50,41 @@ type DHParamConfig struct {
 }
 
 // NewLinkConfig constructs a config from a Frame.
-func NewLinkConfig(frame staticFrame) (*LinkConfig, error) {
+// NOTE: this currently only works with Frames of len(DoF)==0 and will error otherwise
+// NOTE: this will not work if more than one Geometry is returned by the Geometries function.
+func NewLinkConfig(frame Frame) (*LinkConfig, error) {
+	if len(frame.DoF()) > 0 {
+		return nil, fmt.Errorf("cannot create link config for Frame with %d DoF", len(frame.DoF()))
+	}
+
+	pose, err := frame.Transform([]Input{})
```

Comment:
Is there no way to get the pose of a frame directly? Why the empty input transform?

Reply:
A frame doesn't "have" a pose, a frame is a function that knows how to turn a set of floats (inputs) into a pose.

Reply:
We could cast it to a (*staticFrame) and then access the variables directly.  But this is slightly more general since there could be other types of frames of zero DoF.  Idk I could go either way on it.  It isn't performance critical so I think this is fine

Reply:
Personally I prefer the current implementation over a cast to a (*staticFrame)